ludwig/schema/utils.py

```diff
@@ -87,20 +88,24 @@ def RegularizerOptions(default: Union[None, str] = None, allow_none: bool = True
     return StringOptions(["l1", "l2", "l1_l2"], default=default, allow_none=allow_none, description=description)
 
 
-def String(default: Union[None, str] = None, allow_none: bool = True, description=""):
+def String(default: Union[None, str] = None, allow_none: bool = True, pattern: str = ".*", description=""):
```

Comment:
@ksbrar for datetime format validation, what do you think about just using the regex pattern validation argument in the `String()` marshmallow field that I added here instead of a new custom marshmallow field?

Reply:
good call! yes this is better, I just double checked that the pattern is plumbed through to the JSON schema automatically so nothing else is needed